HopsanCore/include/HopsanTypes.h

```diff
@@ -160,73 +168,70 @@ class HVector
     //! @brief Clear the array
     void clear()
     {
-        if (mpDataArray)
-        {
-            delete[] mpDataArray;
-            mSize = 0;
-            mpDataArray=0;
+        free(mpDataArray);
+        mpDataArray = 0;
+        mSize = 0;
+        mCapacity = 0;
+    }
+
+    //! @brief Reserve capacity for the array
+    //! @details If new size is smaller than old nothing is changed
+    //! If new size is larger than old, the additional elements will be uninitialized
+    //! @param [in] s New size
+    void reserve(size_t s)
+    {
+        if (s > mCapacity) {
+            void* pNewData = realloc(static_cast<void*>(mpDataArray), s*sizeof(T));
+            if (pNewData) {
+                mpDataArray = static_cast<T*>(pNewData);
+                mCapacity = s;
+            }
+            mSize = std::min(mCapacity, mSize);
         }
     }
 
     //! @brief Resize the array, keeping old data if any.
-    //! @details If new size is smaller than old, old data will be truncated
+    //! @details If new size is smaller than old, old data will intact up to the new size
     //! If new size is larger than old, the additional elements will be uninitialized
     //! @param [in] s New size
-    void resize(const size_t s)
+    void resize(size_t s)
     {
-        // Create new dummy array
-        T* pNewArray = new T[s];
-
-        // Check how many elements to copy
-        size_t nCopy = s;
-        if (size() < s)
-        {
-            nCopy = size();
+        if (s <= mCapacity) {
+            mSize = s;
         }
-
-        // Copy old data to new array
-        for (size_t i=0; i<nCopy; ++i)
-        {
-            pNewArray[i] = mpDataArray[i];
+        else {
+            // Reserve twice the memory, to avoid reallocation on every resize in append case
+            //! @todo Multiplying with 2 may be OK for small arrays, but for very large ones this could be a problem
```

Comment:
Is multiplication by two a state-of-the-art solution used by other libraries etc?

Reply:
I dont know, someone told me that is what the standard library in gcc does, but I have not been able to verify that, should be quite easy now that I think about it. Maye will do at some point. 
I could not think of any other clever solution here. But if you know the size of the data before hand, then reserve is preferred.
I was also considering adding a preallocation of maybe 10, to speed up resize for small arrays, but decided not to.